.devcontainer/Dockerfile

```diff
@@ -0,0 +1,21 @@
+FROM gcr.io/envoy-ci/envoy-build:04f06115b6ee7cfea74930353fb47a41149cbec3
+
+ARG USERNAME=vscode
```

Comment:
I think what is here to start with is fine. I think maybe add a TODO around migrating this to the actual UID outside the container (which is how `run_envoy_docker.sh` works and matches Linux ownership semantics if you agree.

Reply:
I added remoteUser/containerUser in devcontainer.json, so I think VSCode will update this?

Reply:
It didn't seem to when I did my last experiment with this, but that was remote Docker on a different host, so maybe that was stretching it. Let's go with this for now and update as we go.